package/src/MobxQuery/MobxQuery.test.ts

```diff
@@ -0,0 +1,264 @@
+    );
+
+    expect(queryA === queryB, 'при разных ключах, квери разные').toBe(false);
+
+    expect(queryA === queryC, 'при одинаковом ключе, квери тот же самый').toBe(
+      true,
+    );
+  });
+
+  it('Проверяем создание сторов с разными fetchPolicy, с одинаковым ключом', async () => {
+    const mobxQuery = new MobxQuery();
+    const queryA = mobxQuery.createInfiniteQuery(['foo'], () =>
+      Promise.resolve([]),
+    );
+    const queryB = mobxQuery.createInfiniteQuery(
+      ['foo'],
+      () => Promise.resolve([]),
+      { fetchPolicy: 'cache-first' },
+    );
+    const queryC = mobxQuery.createInfiniteQuery(
+      ['foo'],
+      () => Promise.resolve([]),
+      { fetchPolicy: 'network-only' },
+    );
+
+    const queryD = mobxQuery.createInfiniteQuery(
+      ['foo'],
+      () => Promise.resolve([]),
+      { fetchPolicy: 'network-only' },
+    );
+
+    expect(
+      queryA === queryB,
+      'проверяем что инстантс квери без политики, это тот же инстанс с "cache-first"',
+    ).toBe(true);
+
+    expect(
+      queryB === queryC,
+      'при разных политиках, инстансы сторов должны быть разные',
+    ).toBe(false);
+
+    expect(
+      queryC !== queryD,
+      'два "network-only" квери с одним ключом, это два разных инстанса',
+    ).toBe(true);
+  });
+
+  it('Проверяем работу инвалидации с простыми ключами', async () => {
+    const mobxQuery = new MobxQuery();
+    const queryAsc = mobxQuery.createInfiniteQuery(
+      ['foo', { direction: 'asc' }],
+      () => Promise.resolve(['foo', 'data']),
+    );
+    const queryDesc = mobxQuery.createInfiniteQuery(
+      ['foo', { direction: 'desc' }],
+      () => Promise.resolve(['data', 'foo']),
+    );
+
+    const queryUser = mobxQuery.createQuery(['user'], () =>
+      Promise.resolve({ name: 'Ваня' }),
+    );
+
+    const queries = [queryAsc, queryDesc, queryUser];
+
+    expect(
+      queryAsc.data,
+      'queryAsc проверяем что данных действительно нет',
+    ).toBe(undefined);
+
+    expect(
+      queryDesc.data,
+      'queryDesc проверяем что данных действительно нет',
+    ).toBe(undefined);
+
+    expect(
+      queryUser.data,
+      'queryUser проверяем что данных действительно нет',
+    ).toBe(undefined);
+
+    // запускаем запрос данных везде
+    queryAsc.sync();
+    queryDesc.sync();
+    queryUser.sync();
+    await when(() => checkLoading(queries));
+
+    expect(
+      queryAsc.data,
+      'queryAsc проверяем что данные появились',
+    ).toStrictEqual(['foo', 'data']);
+
+    expect(
+      queryDesc.data,
+      'queryDesc проверяем что данные появились',
+    ).toStrictEqual(['data', 'foo']);
+
+    expect(
+      queryUser.data,
+      'queryUser проверяем что данные появились',
+    ).toStrictEqual({ name: 'Ваня' });
+
+    // запускаем инвалидацию по одной вариации ключа
+    mobxQuery.invalidate(['user']);
+
+    expect(
+      queryAsc.data,
+      'queryAsc дергаем дату у всех сторов, что бы тригернуть загрузку',
```

Comment:
тут и у двух ожиданий ниже
```suggestion
      'queryAsc дергаем дату у всех сторов, чтобы тригернуть загрузку',
```